numpy/core/src/multiarray/compiled_base.c

```diff
@@ -80,6 +80,349 @@ minmax(const npy_intp *data, npy_intp data_len, npy_intp *mn, npy_intp *mx)
     *mx = max;
 }
 
+NPY_NO_EXPORT PyObject *
+arr_histogram_uniform(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
+{
+    PyObject *obj_x, *obj_edges, *weights = Py_None;
+    PyArrayObject *arr_x = NULL, *arr_edges = NULL, *arr_w = NULL, *hist = NULL;
+    npy_intp *ihist = NULL, bins, len, i, ix;
+    double *x = NULL, *e = NULL, *w = NULL, *dhist = NULL, min, max, norm, tx;
+
+    static char *kwlist[] = {"x", "edges", "weights", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|O:_histogram_uniform",
+            kwlist, &obj_x, &obj_edges, &weights)) {
+        goto fail;
+    }
+
+    arr_x = (PyArrayObject *)PyArray_ContiguousFromAny(obj_x, NPY_DOUBLE, 1, 1);
+    if (arr_x == NULL) {
+        PyErr_SetString(PyExc_TypeError, "Couldn't parse the input array.");
```

Comment:
when array constructors fail they already set an error message, I don't think extra information is needed here.